actions_test.go

```diff
@@ -0,0 +1,299 @@
+	// Act
+	triggers, err := client.ListTriggerDefinitions(nil)
+	// Assert
+	autopilot.Ok(t, err)
+	autopilot.Equals(t, 2, triggers.TotalCount)
+	autopilot.Equals(t, "Release", triggers.Nodes[1].Name)
+	autopilot.Equals(t, "Uses Ruby", triggers.Nodes[1].Filter.Name)
+	autopilot.Equals(t, "123456789", triggers.Nodes[1].Owner.Id)
+}
+
+func TestUpdateTriggerDefinition(t *testing.T) {
+	//Arrange
+	request := `{"query":
+		"mutation($input:CustomActionsTriggerDefinitionUpdateInput!){customActionsTriggerDefinitionUpdate(input: $input){triggerDefinition{{ template "custom_actions_trigger_request" }},errors{message,path}}}",
+		"variables":{"input":{"id":"123456789", "filterId":null}}
+	}`
+	response := `{"data": {"customActionsTriggerDefinitionUpdate": {
+      "triggerDefinition": {{ template "custom_action_trigger1" }},
+      "errors": []
+  }}}`
+
+	client := ABetterTestClient(t, "custom_actions/update_trigger", request, response)
+	// Act
+	trigger, err := client.UpdateTriggerDefinition(ol.CustomActionsTriggerDefinitionUpdateInput{
+		Id:     "123456789",
+		Filter: ol.NewID(""),
+	})
+	// Assert
+	autopilot.Ok(t, err)
+	autopilot.Equals(t, "Release", trigger.Name)
+}
+
+func TestUpdateTriggerDefinition2(t *testing.T) {
+	//Arrange
+	request := `{"query":
+		"mutation($input:CustomActionsTriggerDefinitionUpdateInput!){customActionsTriggerDefinitionUpdate(input: $input){triggerDefinition{{ template "custom_actions_trigger_request" }},errors{message,path}}}",
+		"variables":{"input":{"id":"123456789", "name":"test", "description": ""}}
+	}`
+	response := `{"data": {"customActionsTriggerDefinitionUpdate": {
+      "triggerDefinition": {{ template "custom_action_trigger1" }},
+      "errors": []
+  }}}`
+
+	client := ABetterTestClient(t, "custom_actions/update_trigger2", request, response)
+	// Act
+	trigger, err := client.UpdateTriggerDefinition(ol.CustomActionsTriggerDefinitionUpdateInput{
+		Id:          "123456789",
+		Name:        ol.NewString("test"),
+		Description: ol.NewString(""),
+	})
+	// Assert
+	autopilot.Ok(t, err)
+	autopilot.Equals(t, "Release", trigger.Name)
+}
+
+func TestDeleteTriggerDefinition(t *testing.T) {
+	//Arrange
+	request := `{"query":
+		"mutation($input:IdentifierInput!){customActionsTriggerDefinitionDelete(input: $input){deletedAlias,deletedId,errors{message,path}}}",
+		"variables":{"input":{"id":"123456789"}}
+	}`
+	response := `{"data": {"customActionsTriggerDefinitionDelete": {
+      "deletedAlias": "",
+      "deletedId": "123456789",
+      "errors": []
+  }}}`
+	responseErr := `{"data": {"customActionsTriggerDefinitionDelete": {
+      "deletedAlias": null,
+      "deletedId": null,
+      "errors": [{{ template "error1" }}]
+  }}}`
+
+	client := ABetterTestClient(t, "custom_actions/delete_trigger", request, response)
+	clientErr := ABetterTestClient(t, "custom_actions/delete_trigger_err", request, responseErr)
+	clientErr2 := ABetterTestClient(t, "custom_actions/delete_trigger_err2", request, "")
```

Comment:
I did some extra error testing here to hit code paths we've never hit before to increase test coverage - we should probably move these somewhere "common" so they are not buried here